lib/sprockets/resolve.rb

```diff
@@ -165,7 +167,9 @@ def resolve_index_under_path(load_path, logical_name, mime_exts)
         if directory?(dirname)
           candidates = find_matching_path_for_extensions(dirname, "index", mime_exts)
         end
-        return candidates, deps
+        return candidates.map { |c|
+          { filename: c[0], type: c[1], index_alias: c[0].sub(/\/index/, '') }
```

Comment:
Is there any case were a string like `assets/index/foo.js` end up in c`[0]`? I think we can use `gsub` here and also use something like `/\/index.\w{2,4}$/`.